src/include/oshmpi_impl.h

```diff
@@ -173,16 +173,35 @@ typedef struct OSHMPI_space_list {
 
 struct OSHMPI_am_pkt;
 
+typedef struct OSHMPI_team {
+    int my_pe;
+    int n_pes;
+    MPI_Comm comm;
+    MPI_Group group;
+    shmem_team_config_t config;
+} OSHMPI_team_t;
+
 typedef struct {
     int is_initialized;
     int is_start_pes_initialized;
-    int world_rank;
-    int world_size;
+    int world_rank;     /* cache of my_pe for SHMEM_TEAM_WORLD */
+    int world_size;     /* cache of n_pes for SHMEM_TEAM_WORLD */
     int thread_level;
     size_t page_sz;
 
+    /* cache of comm, group for SHMEM_TEAM_WORLD */
     MPI_Comm comm_world;        /* duplicate of COMM_WORLD */
     MPI_Group comm_world_group;
+    shmem_team_config_t team_world_config;
+    OSHMPI_team_t *team_world;  /* cache a team object for easier code in split */
```

Comment:
I think we can skip caching of ` shmem_team_config_t team_world_config`, same for the other predefined teams. It is not set by user.

Reply:
Maybe it is better to keep the naming of these cached items consistent. How about we rename:
`world_rank|rank` -> `team_world_my_pe|n_pes`
`comm_world`->`team_world_comm`
`comm_world_group`->`team_world_group `